uo_sys_framework/AOLimitModule/cfgFunctions/common/fn_AOLimitInit.sqf

```diff
@@ -6,177 +6,175 @@ UO_FW_EXEC_CHECK(CLIENT);
 
-	if ((count ((missionNamespace getvariable _arrayname))) isEqualto 1) then {
-		if ((vehicle _unit) inArea _area) then {
-			_startedInside = true;
-			_outSide = false;
-			_enteredZone = true;
-		} else {
-			if !(_entryMode) exitwith {
-				_run = false;
-				//diag_log "exiting AO limit fnc, already one running!";
-				breakOut "AOLimitMainSpawn";
-			};
-			_startedInside = false;
-			_outSide = true;
-			_enteredZone = false;
-		};
-	} else {
-		if (({(vehicle _unit) inArea _x} count ((missionNamespace getvariable _arrayname))) > 0) then {
-			_startedInside = true;
-			_outSide = false;
-			_enteredZone = true;
-		} else {
-			if !(_entryMode) exitwith {
-				_run = false;
-				//diag_log "exiting AO limit fnc, already one running!";
-				breakOut "AOLimitMainSpawn";
-			};
-			_startedInside = false;
-			_outSide = true;
-			_enteredZone = false;
-		};
-	};
+    LOG_3("unit starting _outSide: %1 _startedInside: %2 _enteredZone: %3",_outSide,_startedInside,_enteredZone);
+    LOG("Starting AO Limit");
 
-	LOG_3("unit starting _outSide: %1 _startedInside: %2 _enteredZone: %3",_outSide,_startedInside,_enteredZone);
-	LOG("Starting AO Limit");
+    if (_run) then {
+        private _AOLimitPFHhandle = [{
+            params ["_argNested", "_idPFH"];
+            _argNested params ["_unit","_args","_startedInside","_outSide","_enteredZone","_arrayname","_softAOMode","_recheckDead","_pos"];
+            _args params ["","_area","","_entryMode","_airsetting","","_softAOtime","_softAOtimeAir"];
 
-	if (_run) then {
-		private _AOLimitPFHhandle = [{
-			params ["_argNested", "_idPFH"];
-			_argNested params ["_unit","_args","_startedInside","_outSide","_enteredZone","_arrayname","_softAOMode","_recheckDead","_pos"];
-			_args params ["_logic","_area","_selectedSides","_entryMode","_airsetting","_AOMode","_softAOtime","_softAOtimeAir"];
+            LOG_1("_argNested: %1",_argNested);
+            LOG_1("_area: %1",_area);
+            LOG_1("(vehicle _unit): %1",(vehicle _unit));
+            private _air = ((vehicle _unit) isKindOf "Air");
+            if ((_airsetting) && {_air}) exitwith {};
 
-			LOG_1("_argNested: %1",_argNested);
-			LOG_1("_area: %1",_area);
-			LOG_1("(vehicle _unit): %1",(vehicle _unit));
-			private _air = ((vehicle _unit) isKindOf "Air");
-			if ((_airsetting) && {_air}) exitwith {};
+            if ((count ((missionNamespace getvariable _arrayname))) isEqualto 1) then {
+                if ((vehicle _unit) inArea _area) then {
+                    LOG("unit in area");
+                    _outSide = false;
+                    _argNested set [3,_outSide];
+                    _enteredZone = true;
+                    _argNested set [4,_enteredZone];
+                    if !(_softAOMode) then {
+                        _pos = getPosATL (vehicle _unit);
+                        _argNested set [8,_pos];
+                    } else {
+                        missionNamespace setVariable ["UO_FW_AOL_Display", _outSide];
```

Comment:
There is no benefit to these macros other than writing these scripts. These are not ACE macros either, they do not take COMPONENT names but simply a UO_FW_ prefix. It's really only there to save you from having to type out UO_FW_ every time you want to get/set a var. Replacing the `nameSpace get/setvariable blah blah` with the macros should be an extremely low priority. This would be very different if modules were in separate pbos with separate configs but they are not.

Reply:
Also to note because these use QUOTE they should not be used with dynamic variable names, such as when you are storing a dynamic varname in a script and compiling it via a getvariable